_build/Extension-SetIdentityAndVersion.ps1

```diff
@@ -0,0 +1,198 @@
+# THIS SCRITP WILL NEED TO BE SPLITED NEXT TIME, GETTING TO LONG.
+[CmdletBinding()]
+Param(
+  [Parameter(Mandatory=$True,Position=1)]
+  [string]$vsixManifestFile,
+
+  [Parameter(Mandatory=$True,Position=2)]
+  [string]$vsixIdentity,
+
+  [Parameter(Mandatory=$True,Position=3)]
+  [string]$vsixDisplayName,
+
+  [Parameter(Mandatory=$True,Position=4)]
+  [string]$buildNumber,
+
+  [Parameter(Mandatory=$True,Position=5)]
+  [string]$vsixCommandMenuName,
+
+  [Parameter(Mandatory=$True,Position=6)]
+  [string]$packageGuid,
+
+  [Parameter(Mandatory=$True,Position=7)]
+  [string]$cmdSetGuid,
+
+  [Parameter(Mandatory=$False,Position=8)]
+  [string]$targetVsixCommandMenuName = "Rapid XAML (local)",
+
+  [Parameter(Mandatory=$False,Position=9)]
+  [string]$targetPackageGuid = "c735dfc3-c416-4501-bc33-558e2aaad8c5",
+
+  [Parameter(Mandatory=$False,Position=10)]
+  [string]$targetCmdSetGuid = "8c20aab1-50b0-4523-8d9d-24d512fa8154"
+)
+
+
+$VersionRegex = "(\d+)\.(\d+)\.(\d+)\.(\d+)"
+
+if($buildNumber -match $VersionRegEx){
+
+  Write-Output "Parsed Date From Build: $dateFromBuildNumber"
+
+  $revision =  [int]::Parse($matches[4]).ToString()
+  
+  $versionNumber = [int]::Parse($matches[1]).ToString() + "." + [int]::Parse($matches[2]).ToString() + "." + [int]::Parse($matches[3]).ToString() + "." + $revision
+  Write-Host "Version Number" $versionNumber
+  
+}
+else{
+	throw "Build format does not match the expected pattern (buildName_w.x.y.z)"
```

Comment:
This example build number in the error message doesn't match the Regex ("0.0.0.0") what does the "buildName_" part mean?

Reply:
The version number is extracted from the build number, that is in format defined as dev.version_$(version.major).$(version.minor).$(Year:yy)$(DayOfYear)$(Rev:.rr), resulting f.ex. in dev.version_0.19.18333.01

Reply:
Ok. It's not obvious that this will match the RegEx though. Sample input(s) might help here too then. Especially if that sample input can be run locally to test the script.